### PR TITLE
Feature/mx 1510 implement backend id provider in extractors

### PR DESCRIPTION
**Pull request by mr-kamran-ali**
- return deterministic static stableTargetId and identifier if queries for "mex" primary source. 
- return a nicer error for wrong query parameters for the fetch endpoint
- update tests

### CODE REVIEW

File: mex/backend/identity/main.py [hunk 1]
```diff
@@ -24,6 +29,17 @@ def assign_identity(request: IdentityAssignRequest) -> Identity:
         raise MExError("found multiple identities indicating graph inconsistency")
     if len(graph_result.data) == 1:
         identity = transform_identity_result_to_identity(graph_result.data[0])
+    elif (
+        request.identifierInPrimarySource
+        == MEX_PRIMARY_SOURCE_IDENTIFIER_IN_PRIMARY_SOURCE
+        and request.hadPrimarySource == MEX_PRIMARY_SOURCE_STABLE_TARGET_ID
```
Comment (carolinscholl) [inline mex/backend/identity/main.py:35]: I don't understand this case, could you please add a comment on this? If I understand correctly, you are comparing to two constants here. I thought we only have the cases of: 
1. ID already in graph -> return that
2. ID not yet in graph -> generate new one

What is this case covering?

**Comment by carolinscholl** [comment]
Looks good to me, I just have a few general questions rather than requests for changes. But main has to be merged.

File: mex/backend/identity/main.py [hunk 2]
```diff
@@ -42,11 +58,14 @@ def fetch_identity(
...
-    graph_result = connector.fetch_identities(
-        had_primary_source=hadPrimarySource,
-        identifier_in_primary_source=identifierInPrimarySource,
-        stable_target_id=stableTargetId,
-    )
+    try:
+        graph_result = connector.fetch_identities(
+            had_primary_source=hadPrimarySource,
+            identifier_in_primary_source=identifierInPrimarySource,
+            stable_target_id=stableTargetId,
```
[the remark this answers is not in the document]
Comment (mr-kamran-ali) [reply mex/backend/identity/main.py:65]: I am not sure if I get it right, "MEX_PRIMARY_SOURCE_STABLE_TARGET_ID" is only used while assigning the stabe_target_id on the basis of provided :
    hadPrimarySource
    identifierInPrimarySource

if these two match "mex" primary source and its stable_target_id (both of these are hardcoded for comparison) then this edge case requires "MEX_PRIMARY_SOURCE_STABLE_TARGET_ID". But request dosnt ask for it. request only requires these two: 
     hadPrimarySource
    identifierInPrimarySource
 request is defined in request model here: https://github.com/robert-koch-institut/mex-backend/blob/64186d47005a40b6a8f9a605f973289ff3035a4c/mex/backend/identity/models.py#L6

[the remark this answers is not in the document]
Comment (carolinscholl) [reply mex/backend/identity/main.py:68]: I think the docstring of fetch should state that you either should provide `stableTargetId` or `hadPrimarySource` and `identifierInPrimarySource` together to get a unique result.

[the remark this answers is not in the document]
Comment (mr-kamran-ali) [reply mex/backend/identity/main.py:72]: good idea, updated in 1c63ebb00470e51bcd9051077f4c827356d205fc

**Review by carolinscholl — APPROVED** [review]
Looks good on the backend part to me!

File: mex/backend/identity/main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cutoffthetop) [reply mex/backend/identity/main.py:72]: to clarify: the fetch endpoint always returns a list of identities.

you need the background-knowledge of how identities work in order to know that providing a filter for `stableTargetId=abc` could give you 0-n results and, conversely, filtering by `hadPrimarySource` and `identifierInPrimarySource` can only give you 0-1 results.

theoretically we don't _have_ to put constraints on which of the 3 filters are combined, we just didn't have a use case yet for only filtering by eg `hadPrimarySource`. that is why the graph connector doesn't implement that yet. see https://github.com/robert-koch-institut/mex-backend/blob/main/mex/backend/graph/connector.py#L174

but you are right, we should put more details in the docs..

**Pull request merged**